src/mlpack/tests/sparse_svm_test.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file sparse_svm_test.cpp
+ * @author Ayush Chamoli
+ *
+ * Test the Sparse SVM class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/sparse_svm/sparse_svm.hpp>
+#include <ensmallen.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::svm;
+using namespace mlpack::distribution;
+
+BOOST_AUTO_TEST_SUITE(SparseSVMTest);
+
+/**
+ * A more complicated test for the SparseSVMFunction.
+ */
+BOOST_AUTO_TEST_CASE(SparseSVMFunctionEvaluate)
+{
+  const size_t dimension = 10;
+  const size_t points = 1000;
+  const size_t trials = 50;
+
+  // Initialize a random dataset.
+  arma::sp_mat data;
+  data = arma::sprandu(dimension, points, 0.2);
+
+  // Create random response.
+  arma::vec responses(points);
```

Comment:
Here also I think we should use `arma::Row<size_t>` (and the labels `0` and `1`) for consistency.